### ISSUE TITLE
PPO Reward Not Increasing

### ISSUE THREAD

**Issue by HenryPengZou** [Wuyxin/collabllm#16]
When running `script/ppo.py` on 4~8 80G A100, it seems that the reward is not increasing (plz see screenshot below). Have the authors encountered a similar situation, or do you have any suggestions for resolving this?
In addition, did you include an SFT (supervised fine-tuning) warm-start stage when training PPO?

I appreciate your time and any thoughts and suggestions on this.

<img width="712" height="356" alt="Image" src="https://github.com/user-attachments/assets/ff0b704a-3eca-4a7e-b9b4-f36af1cbb3bd" />

Here is my command: 
```bash
ENABLE_COLLABLLM_LOGGING=0 LLM_USE_V1=1 VLLM_ENABLE_V1_MULTIPROCESSING=0 WANDB__SERVICE_WAIT=300 CUDA_VISIBLE_DEVICES=4,5,6,7 \
    torchrun --master_port=56600 --nnodes=1 --nproc_per_node=4 -m scripts.train.ppo \
    --dataset_name math-hard \
    --metric_names "accuracy" "token_amount" \
    --metric_weights 1 -0.5 \
    --user_generation_kwargs '{"model": "gpt-4o-mini"}' \
    --assistant_generation_kwargs '{"model": "sft-math-hard-Llama-3.1-8B-Instruct", "temperature": 0.6, "max_tokens": 2048}' \
    --reward_generation_kwargs '{"model": "gpt-4o"}' \
    --dataset_repo collabllm/collabllm-multiturn-math-hard \
    --model_name outputs/sft/collabllm-multiturn-math-hard \
    --base_model_name meta-llama/Llama-3.1-8B-Instruct \
    --output_dir outputs/ppo/collabllm-multiturn-math-hard_llama \
    --batch_size 1 \
    --mini_batch_size 1 \
    --num_ppo_epochs 1 \
    --gradient_accumulation_steps 1 \
    --save_total_limit 10 \
    --num_train_epochs 5 \
    --learning_rate 2e-6 \
    --gpu_memory_utilization 0.2 \
    --logging_steps 1 \
    --wandb_entity treeforest \
    --wandb_project interagent_ppo_llama \
    --num_samples 3 \
    --max_new_turns 4 \
    --max_metric_workers 2 \
    --use_lora \
    --use_4bit
```

I also tried to only set "accuracy" in the "metric_names", but the reward is still not increasing.

<img width="1341" height="730" alt="Image" src="https://github.com/user-attachments/assets/c0b7ba69-a4fc-415a-8be1-bd9870294ace" />

**Comment by Wuyxin** [comment]
Hi 

Yes we did do SFT before PPO. 

The non-increasing reward is likely because the total batch size= gradient_accumulation_steps* batch_size*num_gpus is too small. If you don't use vllm [here](https://github.com/Wuyxin/collabllm/blob/main/scripts/train/ppo.py#L378C42-L378C67), and instead use [_batch_generate_with_huggingface](https://github.com/Wuyxin/collabllm/blob/main/collabllm/simulation.py#L177) or a smaller model, you can set the batch size to be much higher. 

Have you tried DPO? You should be able to see the reward increase properly.

A side note, if you have used verl, we are gonna integrate CollabLLM to verl within these two days, their implementation makes it much eaiser to set a higher batch size than trl.

**Comment by HenryPengZou** [comment]
Hi @Wuyxin, thanks so much for your quick response and helpful suggestion!

I’ve adapted the code on my dataset to VERL, and it works great—the reward is steadily increasing, and I can significantly scale up the batch_size/mini_batch_size. VERL also does an excellent job utilizing GPU memory and speeding up the training process.

I haven’t tried DPO yet but plan to give it a go in the next few days.

Thanks again!!
